optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -169,12 +171,22 @@ public Boolean getDroolsAlphaNetworkCompilationEnabled() {
         return droolsAlphaNetworkCompilationEnabled;
     }
 
+    @Deprecated
     public KieRuntimeBuilderWrapper getGizmoKieRuntimeBuilderWrapper() {
-        return gizmoKieRuntimeBuilderWrapper;
+        return (KieRuntimeBuilderWrapper) getKieBaseSupplier();
     }
 
+    @Deprecated
     public void setGizmoKieRuntimeBuilderWrapper(KieRuntimeBuilderWrapper kieRuntimeBuilderWrapper) {
-        this.gizmoKieRuntimeBuilderWrapper = kieRuntimeBuilderWrapper;
+        setKieBaseSupplier(kieRuntimeBuilderWrapper);
+    }
+
+    public Supplier<KieBase> getKieBaseSupplier() {
+        return kieBaseSupplier;
+    }
+
+    public void setKieBaseSupplier(Supplier<KieBase> kieBaseSupplier) {
+        this.kieBaseSupplier = kieBaseSupplier;
     }
```

Comment:
This could use a Javadoc describing its purpose. 
Specifically, that people really don't need to use this at all.